### PR TITLE
chore: improve clean up pipeline

### PR DESCRIPTION
**Pull request by vilit1**
Ensure clean up pipeline also purges keyvaults. Purge allows us to reuse keyvault names (and avoid clashing between pipeline runs).

To allow purging, the no-wait for keyvault deletion has been removed. 

The general clean-up is kept as a backup for missed resources from `az iot ops delete` (ex: the command fails).
 
 Keyvaults after the cleanup pipeline:
image

 Note: the extra keyvault prefix input is temporary and will be removed once we have all keyvaults using the same one again (we had to diverge and use different ones due to clashes)

Note 2: Since the purge has a --no-wait, the purge may take some time and the results are not instantaneous.

---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to Azure IoT Operations tooling!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

Intent for Production

- [ ] It is expected that pull requests made to default or core branches such as `dev` or `main` are of production grade. Corollary to this, any merged contributions to these branches may be deployed in a public release at any given time. By checking this box, you agree and commit to the expected production quality of code.

Basic expectations

- [ ] If introducing new functionality or modified behavior, are they backed by unit and/or integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** the relevant unit **and** integration tests pass? i.e. `pytest <project root> -vv`. Please provide evidence in the form of a screenshot showing a succesful run of tests locally OR a link to a test pipeline that has been run against the change-set.
- [ ] Have linter checks passed using the `.pylintrc` and `.flake8` rules? Look at the CI scripts for example usage.
- [ ] Have extraneous print or debug statements, commented out code-blocks or code-statements (if any) been removed from the surface area of changes?
- [ ] Have you made an entry in HISTORY.rst which concisely explains your user-facing feature or change?

Azure IoT Operations CLI maintainers reserve the right to enforce any of the outlined expectations.

A PR is considered **ready for review** when all basic expectations have been met (or do not apply).

### CODE REVIEW

File: .github/workflows/cluster_cleanup.yml [hunk 1]
```diff
@@ -30,14 +30,18 @@ on:
...
+      keyvault_prefix:
+        type: string
+        description: "Prefix of keyvault to delete"
+        default: "opskv"
+        required: false
   # Run every night at midnight (Pacific) to cleanup resources
-  schedule:
-    - cron: '0 8 * * *'
+  # schedule:
+  #   - cron: '0 8 * * *'
```
Comment (c-ryan-k) [inline .github/workflows/cluster_cleanup.yml:40]: should we still run this on a schedule?

File: .github/workflows/cluster_cleanup.yml [hunk 2]
```diff
@@ -71,8 +75,13 @@ jobs:
           subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
       - name: Delete keyvaults
         run: |
-          for vault in $(az keyvault list --query "[?starts_with(name, '${{ env.KEYVAULT_PREFIX }}')].name" -o tsv); do
-            az keyvault delete -n $vault -g ${{ env.RESOURCE_GROUP }} --no-wait
+          for vault in $(az keyvault list --query "[?starts_with(name, '${{ inputs.keyvault_prefix }}')].name" -o tsv); do
```
Comment (c-ryan-k) [inline .github/workflows/cluster_cleanup.yml:78]: This wasn't in the previous command for some reason, but I'm thinking we should probably limit the list results to our particular resource group

File: .github/workflows/cluster_cleanup.yml [hunk 3]
```diff
@@ -71,8 +75,13 @@ jobs:
...
+            az keyvault delete -n $vault -g ${{ env.RESOURCE_GROUP }}
+          done
+      - name: Purge keyvaults
+        run: |
+          for vault in $(az keyvault list-deleted --query "[?starts_with(name, '${{ inputs.keyvault_prefix }}')].name" -o tsv); do
```
Comment (c-ryan-k) [inline .github/workflows/cluster_cleanup.yml:83]: same here re: resource group

File: .github/workflows/cluster_cleanup.yml (hunk 1 above)
Comment (vilit1) [reply .github/workflows/cluster_cleanup.yml:40]: yea - I will update that right before we merge it in

File: .github/workflows/cluster_cleanup.yml (hunk 2 above)
Comment (vilit1) [reply .github/workflows/cluster_cleanup.yml:78]: list has an easy -g but list-deleted does not D:

Dammit will need to make more jmes queries

File: .github/workflows/cluster_cleanup.yml (hunk 3 above)
Comment (vilit1) [reply .github/workflows/cluster_cleanup.yml:83]: see above

**Review by c-ryan-k — APPROVED** [review]

**Pull request merged**